tests/libvmi/vmi.go

```diff
@@ -88,9 +88,40 @@ func WithRng() Option {
 // WithResourceMemory specifies the vmi memory resource.
 func WithResourceMemory(value string) Option {
 	return func(vmi *v1.VirtualMachineInstance) {
-		vmi.Spec.Domain.Resources.Requests = k8sv1.ResourceList{
-			k8sv1.ResourceMemory: resource.MustParse(value),
+		if vmi.Spec.Domain.Resources.Requests == nil {
+			vmi.Spec.Domain.Resources.Requests = k8sv1.ResourceList{}
 		}
+		vmi.Spec.Domain.Resources.Requests[k8sv1.ResourceMemory] = resource.MustParse(value)
+	}
+}
+
+// WithResourceCpu specifies the vmi CPU resource.
+func WithResourceCpu(value string) Option {
```

Comment:
We should probably create a new file named `resources` and place these there. The `vmi.go` is increasing too much.
If you have time, it will be nice to see a PR moving there the existing ones (can be a follow up PR).

Reply:
Good idea, I will do it in a future PR.